File: day01/python/part2.py
#!/usr/bin/env python3

from typing import List


def read_lines(fname: str) -> List[str]:
    with open(fname) as f:
        return f.read().strip().splitlines()


def fuel(mass: int) -> int:
    return (mass // 3) - 2


def extra_fuel(mass: int) -> int:
    value = mass
    total = 0

    while value > 0:
        value = fuel(value)
        if value > 0:
            total += value
    #
    return total


def main():
    lines = read_lines("input.txt")
    result = sum(extra_fuel(int(line)) for line in lines)
    print(result)

##############################################################################

if __name__ == "__main__":
    main()

File: day01/c/part2.c
#include "helper.h"

#include <stdio.h>
#include <stdlib.h>

int fuel(const int mass) {
    return (mass / 3) - 2;
}

int extra_fuel(const int mass)
{
    int value = mass;
    int total = 0;

    while (value > 0) {
        value = fuel(value);
        if (value > 0) {
            total += value;
        }
    }
    return total;
}

int main()
{
    char* *lines = read_lines("input.txt");

    int total = 0;
    for (int i = 0; i < arrlen(lines); ++i) {
        char* line = lines[i];
        int value = atoi(line);
        total += extra_fuel(value);
    }
    printf("%d\n", total);

    return 0;
}
